src/app/backend/events.go

```diff
@@ -72,8 +73,27 @@ func GetEvents(client *client.Client, namespace string) (*Events, error) {
 		Namespace: namespace,
 	}
 
-	for _, element := range list.Items {
-		events.Events = append(events.Events, Event{
+	// Get events for replica set.
+	selector, _ := fields.ParseSelector("involvedObject.name=" + replicaSet.Name)
```

Comment:
Try to remove "involvedObject.name" magic string. See how kubectl does this kubernetes/pkg/kubectl/describe.go

```go
func (d *PodDescriber) Describe(namespace, name string) (string, error) {
	rc := d.ReplicationControllers(namespace)
	pc := d.Pods(namespace)

	pod, err := pc.Get(name)
	if err != nil {
		eventsInterface := d.Events(namespace)
		selector := eventsInterface.GetFieldSelector(&name, &namespace, nil, nil)
		options := unversioned.ListOptions{FieldSelector: unversioned.FieldSelector{selector}}
		events, err2 := eventsInterface.List(options)
		if err2 == nil && len(events.Items) > 0 {
			return tabbedString(func(out io.Writer) error {
				fmt.Fprintf(out, "Pod '%v': error '%v', but found events.\n", name, err)
				DescribeEvents(events, out)
				return nil
			})
		}
		return "", err
	}
```